### PR TITLE
ATECC608A flow

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

**Review by robinbudd — APPROVED** [review]

File: implementations/c/external/cryptoauth/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+if (NOT OCKAM_ENABLE_ATECC608A_BUILD)
```
Comment (mrinalwadhwa) [inline implementations/c/external/cryptoauth/CMakeLists.txt:1]: @SanjoDeundiak what was the reason to move cryptoauthlib to cryptoauth?
I wonder if this breaks the ability to find and externally installed cryptoauthlib

File: implementations/c/ockam/key_agreement/xx/tests/init_vault.c [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#include "cryptoauthlib.h"
+#include "atca_cfgs.h"
+#include "atca_iface.h"
+#include "atca_device.h"
+
+ATCAIfaceCfg test_atecc608a_cfg =
+        {
+                .iface_type = ATCA_I2C_IFACE,
+                .devtype = ATECC608A,
+                .atcai2c.slave_address = 0xC0,
```
Comment (nilslice) [inline implementations/c/ockam/key_agreement/xx/tests/init_vault.c:18]: I've created an issue to track the possibility of using more inclusive language for this in the upstream: https://github.com/MicrochipTech/cryptoauthlib/issues/193

File: implementations/c/external/cryptoauth/CMakeLists.txt (hunk 1 above)
Comment (SanjoDeundiak) [reply implementations/c/external/cryptoauth/CMakeLists.txt:1]: @mrinalwadhwa even though header is called cryptoauthlib.h, the original target name from Microchips cmake files is cryptoauth. Honestly, I'm not 100% sure it's the right thing to do, we still have some occasional problems with this dependency which should be investigated in the future, so eventually we're gonna come back to this question.

Comment (mrinalwadhwa) [reply implementations/c/external/cryptoauth/CMakeLists.txt:1]: cool, let's log an issue so we remember to test that externally installed libs should continue to work.

**Review by mrinalwadhwa — DISMISSED** [review]
@SanjoDeundiak this looks ready to merge. Is there someway we can make the commit message more descriptive of the changes in this PR?

File: implementations/c/external/cryptoauth/CMakeLists.txt (hunk 1 above)
Comment (SanjoDeundiak) [reply implementations/c/external/cryptoauth/CMakeLists.txt:1]: @mrinalwadhwa we have them
https://github.com/ockam-network/ockam/issues/351
https://github.com/ockam-network/ockam/issues/291

**Comment by SanjoDeundiak** [comment]
> @SanjoDeundiak this looks ready to merge. Is there someway we can make the commit message more descriptive of the changes in this PR?

Is it better now?

**Review by mrinalwadhwa — APPROVED** [review]

**Pull request merged**